### PR TITLE
#2 Create Initial Design Document

### LINKED ISSUE #2: Create Design Docs
# Overview

Create a design doc for the MVP version of Glitch Diary

### PR DESCRIPTION
**Pull request by xyzarivera**
Resolves #2 

This PR 
- creates the initial design document of the project
- adds a pull request template for the project
- adds an issue template for the project

### CODE REVIEW

File: docs/design-doc.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+### As a user
+- I should be able to bring the Raspberry Pi and use the app anywhere
+- I should be able to submit a log entry
+- I should be able to see the list of logs
+
+### As an engineer
+
+- I should be able to develop an application without using 
+frontend libraries (ReactJS, Vue, Sveltte) or meta-frameworks (NextJS, NuxtJS,
+Sveltte Kit)
```
Comment (nasvillanueva) [inline docs/design-doc.md:27]:
```suggestion
Svelte Kit)
```

typo on svelte spelling

File: docs/design-doc.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  - persist the new log in a storage
+  - read all saved logs
+
+## Scope limitations
+These features are not yet covered:
+  - interface design
+  - reading individual logs
+  - updating logs
+  - deleting logs
+  - testing
```
Comment (nasvillanueva) [inline docs/design-doc.md:54]: Testing should be included. It shouldn’t be difficult to add test for simple api functions

**Review by nasvillanueva — APPROVED** [review]
Feel free to merge once you resolve the comments.

LGTM

File: docs/design-doc.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+### As a user
+- I should be able to bring the Raspberry Pi and use the app anywhere
+- I should be able to submit a log entry
+- I should be able to see the list of logs
+
+### As an engineer
+
+- I should be able to develop an application without using 
+frontend libraries (ReactJS, Vue, Sveltte) or meta-frameworks (NextJS, NuxtJS,
```
Comment (xyzarivera) [inline docs/design-doc.md:26]:
```suggestion
frontend libraries (ReactJS, Vue, Svelte) or meta-frameworks (NextJS, NuxtJS,
```

File: docs/design-doc.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  - persist the new log in a storage
+  - read all saved logs
+
+## Scope limitations
+These features are not yet covered:
+  - interface design
+  - reading individual logs
+  - updating logs
+  - deleting logs
+  - testing
```
Comment (xyzarivera) [inline docs/design-doc.md:54]:
```suggestion
```

File: docs/design-doc.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+The user interface will be developed using HTML, CSS, and JS/TS.
+
+The server will be developed using the ExpressJS framework with a PostgreSQL
+database and REST API.
+
+## Deployment
+
+Both user interface and server will be deployed in a Raspberry Pi.
+
```
Comment (xyzarivera) [inline docs/design-doc.md:65]:
```suggestion

## Testing
Unit test are to be written using Jest.

```

**Pull request merged**